### ISSUE TITLE
Building solution in Visual Studio 2017

### ISSUE THREAD

**Issue by jeronm** [HearthSim/Hearthstone-Deck-Tracker#3224]
I'm sorry if this is a very basic question, but I'm having problems building the main HDT solution in Visual Studio 2017. I am trying to work on the plugin tutorial, but am unable to build a development version of HDT. I'm sure I am struggling with some basic fundamentals of building a project, but the I can't seem to find a clear set of instructions on this. So far I have:

Cloned a copy of the master repo to my hard drive
Ran the bootstrap.bat batch file
Opened 'Hearthstone Deck Tracker.sln' in visual studio
Attempted to build project

I receive many errors because the libraries cannot be found (HearthDb.dll, HearthMirror.dll, etc.). Again, I apologize for my ignorance here, but would it be possible to get a little more clear instruction on how to download and build a new version of the program?

**Comment by riQQ** [comment]
Sounds like there were errors running the `bootstrap.bat`. What's the console output when running the batch?

**Comment by azeier** [comment]
What's the output when you run the bootstrap file?

**Comment by jeronm** [comment]
The first time I run the batch file, I get this output:

'nuget' is not recognized as an internal or external command,
operable program or batch file.
Cloning into 'HearthDb'...
remote: Counting objects: 453, done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 453 (delta 0), reused 0 (delta 0), pack-reused 444
Receiving objects: 100% (453/453), 1.62 MiB | 0 bytes/s, done.
Resolving deltas: 100% (248/248), done.
Cloning into 'HearthMirror'...
remote: Counting objects: 927, done.
remote: Compressing objects: 100% (44/44), done.
remote: Total 927 (delta 17), reused 0 (delta 0), pack-reused 882
Receiving objects: 100% (927/927), 247.99 KiB | 0 bytes/s, done.
Resolving deltas: 100% (575/575), done.
Cloning into 'HSReplay-Api'...
remote: Counting objects: 133, done.
remote: Compressing objects: 100% (8/8), done.
Receiving objects:  48% (64/133)    0 (delta 0), pack-reused 125
Receiving objects: 100% (133/133), 46.85 KiB | 0 bytes/s, done.
Resolving deltas: 100% (72/72), done.
Cloning into 'HDT-Localization'...
remote: Counting objects: 144, done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 144 (delta 6), reused 0 (delta 0), pack-reused 132
Receiving objects: 100% (144/144), 237.04 KiB | 0 bytes/s, done.
Resolving deltas: 100% (87/87), done.
HDT-Localization\Strings.de-DE.resx
HDT-Localization\Strings.fr-FR.resx
HDT-Localization\Strings.ja-JP.resx
HDT-Localization\Strings.ko-KR.resx
HDT-Localization\Strings.pl-PL.resx
HDT-Localization\Strings.pt-BR.resx
HDT-Localization\Strings.pt-PT.resx
HDT-Localization\Strings.resx
HDT-Localization\Strings.ru-RU.resx
HDT-Localization\Strings.uk-UA.resx
HDT-Localization\Strings.zh-CN.resx
HDT-Localization\Strings.zh-TW.resx
12 File(s) copied
'msbuild' is not recognized as an internal or external command,
operable program or batch file.
The system cannot find the path specified.

I'm not sure I fully understand the purpose of 'nuget'. When I read the instructions, it said it was included in Visual Studio 2017 so I did not download any additional files.

**Comment by riQQ** [comment]
Run the batch from the `Developer Command Prompt for VS 2017`.

**Comment by jeronm** [comment]
I think I have it working now. It seemed to be a directory path issue and it broke at a point where I had a space in a folder name. I'm not sure if it was the depth of the directory or the space, but I found an error that didn't have the correct path.

When I ran the batch file in the developer command prompt, I get the following error:

Program.cs(10,7): error CS0246: The type or namespace name 'nQuant' could not be found (are you missing a using directive or an assembly reference?) [D:\Users\XX\Desktop\Hearthstone-Deck-Tracker-master\ResourceGenerator\ResourceGenerator.csproj]

I then loaded the solution in Visual Studio and the build worked. Not sure if that error is important, but I think everything is working now.

Thanks for the help!
